lusSTR/annot.py

```diff
@@ -764,3 +903,31 @@ def main(args):
         print(summary, file=output_file)
         if args.out is not None:
             output_file.close()
+
+        if not args.uas and args.kit == "forenseq":
+            if flank_5_anno == "":
+                full_bracketed_anno = f"{forward_strand_bracketed_form} {flank_3_anno}"
+            elif flank_3_anno == "":
+                full_bracketed_anno = f"{flank_5_anno} {forward_strand_bracketed_form}"
+            else:
+                full_bracketed_anno = (
+                    f"{flank_5_anno} {forward_strand_bracketed_form} {flank_3_anno}"
+                )
+            flank_summary = [
+                sampleid, project, analysis, locus, reads, str_allele, sequence, flank_5_anno,
+                forward_strand_bracketed_form, flank_3_anno
+            ]
+            flank_summary = '\t'.join(str(i) for i in flank_summary)
+            name = re.sub(".txt", "", args.out)
+            if os.path.exists(f"{name}_flanks_anno.txt"):
+                flank_file = open(f"{name}_flanks_anno.txt", "a")
+            else:
+                flank_file = open(f"{name}_flanks_anno.txt", "w")
```

Comment:
I think you can get away with just using `open(filename, 'a')` here—it will create the output file if it doesn't yet exist. No need for the if/else statement.

On a different note, I would strongly recommend against creating any output files that the user doesn't provide a name for. In this case, I'd recommend only printing the flank summary if the user requests it using some flag like `-f|--flank`.

Is there any reason the flank data shouldn't/couldn't be included in the normal output? I'm not saying this is something we should look into for this PR, but it may be worth considering including variation in flanking regions in the default output (or possibly extra columns in the normal output if requested by the user).

Reply:
Yeah, so this was something I talked with RebJ about. We thought including this information in the original table is a little too much, especially since some of these flanks are pretty long. I had suggested to RebJ about adding a flag option to invoke the flanks report and she thought requiring people to add more optional flags would be challenging for people. She's on the perspective of normal forensics people who have little to no coding experience. She said that she would prefer to have multiple reports outputted at the same time and she would decide whether to keep them or not. What are your thoughts? I also had concerns about relying on the user to provide the "correct" filename (i.e. a .txt file).

Reply:
Fair point about piling too many columns into a single output file. I suspected that might be an issue.

As far as how to name the output file: it shouldn't be too complicated to come up with a output filename prefix by stripping off the input filename suffix in an extension-agnostic manner. I still think it's a suboptimal design choice, but I can think of many far worse sins one could commit when trying to accommodate novice users.

A little venting on that last note (sorry Jacki!): I've worked with many beginners (biology or other non-computer backgrounds) teaching basic shell/terminal skills. Operating in the shell has a learning curve for everyone. Some folks just take it as a matter of course. Other folks seem unwilling to put the same amount of effort into learning the shell that they put into learning new lab techniques, SOPs, legal requirements, or a host of other things that are at least as tedious and sophisticated as following explicit instructions on how to run a command in the terminal. I eventually learned that the former group deserved and appreciated any extra attention I could give them to clarify and simplify the content, but that most efforts aimed at the latter group were in vain.

Reply:
At some point, we may want to throw a "simple" GUI with graphic file selectors onto lusSTR to make it more accommodating. (Being able to create isolated environments to install and test lusSTR on a Windows would be helpful here.) But until we're ready to make that investment in time/effort/testing/maintenance, I think we need to make it clear that lusSTR is a command line tool, and anyone wanting to use it should expect to need hand holding unless they're willing and able to learn command line basics.

Reply:
But back to your comment on using only `open(filename, 'a')`.... that won't create the header then?

Reply:
Heh. You're right. Skimmed over that detail in my first read through.

Reply:
Let me take a crack at the filename thing.

Reply:
I have the extension part done. Pushing now.

Reply:
i.e. stripping the file extension off agnostically.

Reply:
Well there you go, beat me to it! 😀